File: Problemas/32B - Borze/32B_Borze.py
# 32B - Borze!
# https://codeforces.com/problemset/problem/32/B

# Input:
# 1) borze = Código Borze
borze = input()

# Convierte '--' a 2
borze_decodificado_2 = borze.replace('--', '2')

# Convierte '-.' a 1 sobre el Borze decodificado con el 2
borze_decodificado_1 = borze_decodificado_2.replace('-.', '1')

# Convierte '.' a 0 sobre el Borze decodificado con el 1
# Esta último conversión devolverá el Borze decodificado final
borze_decodificado = borze_decodificado_1.replace('.', '0')

# Imprime el código Borze decodificado
print(borze_decodificado)

File: Problemas/32B - Borze/32B_Borze.cpp
// 32B - Borze
// https://codeforces.com/problemset/problem/32/B

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) borze = Código Borze
    string borze;
    cin >> borze;

    // Longitud del código
    int longitud_codigo = borze.length();

    // Itera sobre los caracteres del código
    for(int caracter = 0; caracter < longitud_codigo; caracter++) {
        
        // Si el caracter es un punto, entonces es un 0
        if(borze[caracter] == '.') {
            cout << "0";

        // Si es caracter no es un punto, entonces es un guión, por lo que es un 1 o 2
        } else { 

            // Suma +1 al índice para pasar al siguiente caracter
            caracter++;

            // Si el caracter es un punto, entonces es un guión+punto que es un 1
            if(borze[caracter] == '.') {
                cout << "1";
            
            // Si es caracter no es un punto, entonces es un guión, por lo que es un guión+guión que es un 2
            } else { 
                cout << "2";
            }
        }
    } 

    return 0;
}